frontend/src/hooks/query/use-user-connected.ts

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from "@tanstack/react-query";
+import OpenHands from "#/api/open-hands";
+
+/**
+ * Hook to check if a user is connected to any provider (including local git provider)
+ * This is determined by whether the /api/user/info endpoint returns a 200 status code
+ */
+export const useUserConnected = () => {
+  return useQuery({
+    queryKey: ["user-connected"],
```

Comment:
```suggestion
    queryKey: ["user", "connected"],
```

Our user-related queries are grouped with the "user" key, it make it easier to invalidate user-related queries at once (e.g., `queryClient.removeQueries({ queryKey: ["user"] });`)